framework/src/org/apache/cordova/CordovaWebViewClient.java

```diff
@@ -89,6 +97,15 @@ public void setWebView(CordovaWebView view) {
         this.appView = view;
         helper = new CordovaUriHelper(cordova, view);
     }
+    
+    /**
+     * Specifies the HTTP auth handler for handling auth challenge requests.
+     *
+     * @param handler HTTP auth handler.
+     */
+    public void setHttpAuthRequestHandler(HttpAuthRequestHandler handler) {
```

Comment:
Andrew could you please clarify the callback idea in more details? - Does it mean walk thru all available plugins and see which one implements some specific interface (like HttpAuthRequestHandler) and if so call corresponding method?